pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -184,6 +246,11 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 	if !exists {
 		return nil, fmt.Errorf("object %q/%q not registered", namespace, name)
 	}
+
+	if !item.isImmutable() && item.isStopped() {
+		item = c.updateItem(key, item)
```

Comment:
this can in theory go back in time, which would be extremely misleading for users
The case is:
- we initially fetched data from apiserver A
- we stopped
- now we refetch from apiserver B which is lagging and has the older state

That will never happen within a single running reflector because the next invocations of list/Watch set min RV. It's not the case for creating a reflector.

Reply:
If we just rerun the old reflector, we do not know when the reflector will synchronize again. The data in the old reflector may be obtained.

Another possibility is that if the reflector is not running for a long time, there is a high probability that the 'too old resource version error' occurs and the reflector will force list-watch again.

Reply:
Run() always starts from List (and "too old RV are handled correctly - they may happen anytime).

Regarding checking when the reflector will synchronize, that's a good question. What you can potentially do is to replace implementation of the underlying store with yours that will redefine the "Replace" operation (which is what the first List is calling) and basically:
- on reinitiating set the "initialized" field to false
- call reflector.Run()
- modify HasSynced to use this field

Reply:
I have modified the code based on the review comments, and add the testcase.